net/interfaces/interfaces_windows.go

```diff
@@ -7,76 +7,96 @@ package interfaces
 import (
 	"fmt"
 	"log"
+	"net"
 	"net/url"
-	"os/exec"
 	"syscall"
 	"unsafe"
 
-	"go4.org/mem"
 	"golang.org/x/sys/windows"
 	"golang.zx2c4.com/wireguard/windows/tunnel/winipcfg"
 	"inet.af/netaddr"
 	"tailscale.com/tsconst"
-	"tailscale.com/util/lineread"
+)
+
+const (
+	fallbackInterfaceMetric = uint32(0) // Used if we cannot get the actual interface metric
 )
 
 func init() {
 	likelyHomeRouterIP = likelyHomeRouterIPWindows
 	getPAC = getPACWindows
 }
 
-/*
-Parse out 10.0.0.1 from:
-
-Z:\>route print -4
-===========================================================================
-Interface List
- 15...aa 15 48 ff 1c 72 ......Red Hat VirtIO Ethernet Adapter
-  5...........................Tailscale Tunnel
-  1...........................Software Loopback Interface 1
-===========================================================================
-
-IPv4 Route Table
-===========================================================================
-Active Routes:
-Network Destination        Netmask          Gateway       Interface  Metric
-          0.0.0.0          0.0.0.0         10.0.0.1       10.0.28.63      5
-         10.0.0.0      255.255.0.0         On-link        10.0.28.63    261
-       10.0.28.63  255.255.255.255         On-link        10.0.28.63    261
-        10.0.42.0    255.255.255.0   100.103.42.106   100.103.42.106      5
-     10.0.255.255  255.255.255.255         On-link        10.0.28.63    261
-   34.193.248.174  255.255.255.255   100.103.42.106   100.103.42.106      5
-
-*/
 func likelyHomeRouterIPWindows() (ret netaddr.IP, ok bool) {
-	cmd := exec.Command("route", "print", "-4")
-	cmd.SysProcAttr = &syscall.SysProcAttr{HideWindow: true}
-	stdout, err := cmd.StdoutPipe()
+	rs, err := winipcfg.GetIPForwardTable2(windows.AF_INET)
 	if err != nil {
+		// TODO: It would be nice to log this...
```

Comment:
we can make all these take a `logger.Logf`. Want to send that first?

Reply:
Yep, I've tried to add `logger.Logf` input argument to the function, but it turned out to be not too simple because the function signature is used in several places, and I didn't want to introduce too much change in this PR. But this is the solution that I prefer, and I believe that we should do that, now or later.

But I've just discovered that there's an alternative. In the darwin counterpart function (`likelyHomeRouterIPDarwinFetchRIB`) you've also needed logging, but there you've implemented it simply by using `log.Printf` function. I still don't know all the ins and outs of the current logging system, but at the first glance in this example it looks a bit inconsistent to me (sometimes using `log.Printf`, sometimes `logger.Logf`).

I'll temporary use `log.Printf` (as the darwin counterpart does), but I would be happy to change the signature to accept `logger.Logf` argument if you agree.

Reply:
That works for now. We have parts of our system (in tests) that detect whether you're bypassing the passed-around logger.Logf and using log.Printf directly and explode the whole program (for catching it in tests). But I guess you (and darwin's impl) are getting lucky here and this isn't exercised during those tests.